### ISSUE TITLE
Incorrect User Role assignments

### ISSUE THREAD

**Issue by twstewart42** [PagerDuty/terraform-provider-pagerduty#170]
Using the pagerduty_user resource and assigning a role to a user. After the apply, the role assigned to the user is different than expected.

It's possible that this is an issue with the API and not terraform provider, but I am unsure.

### Terraform Version
.\terraform.exe -v
Terraform v0.12.17
+ provider.pagerduty v1.4.1

### Affected Resource(s)
The user is assigned to an incorrect role.

### Terraform Configuration Files
```
resource "pagerduty_user" "user123" {
  name        = "User One"
  email       = "user123@example.com"
  role        = "user"
}
```

### Expected Behavior vs Actual Behavior
After a lot of trial and error to get it figured out. Here is the general mapping I worked out while experimenting.  
**Terraform            vs Pageduty**
limited_user       == Responder 
team_responder == Observer 
user                    == Manager
admin                 == Global Admin

I would not expect an individual with the user role, to be giving Manager level access. 
I would expect an individual assigned as a team_responder, could do more than observe

### Important Factoids
https://www.terraform.io/docs/providers/pagerduty/r/user.html#role

**Comment by h3x89** [comment]
I would like to add that there is option for advance permissions for users:
https://support.pagerduty.com/docs/advanced-permissions#section-base-roles

Is there any plans to support that?

**Comment by dbamohsin** [comment]
Thank you @twstewart42  - this was giving me a headache as the correlations made no sense at all!

**Comment by matt-canty-dragon** [comment]
@twstewart42 what you're seeing there might not even be the API, but how it has been designed in the backend. I also do not find this desirable as we expect our SAML integration to generate users on first log on. The problem with this is that with advanced permissions they are given Manager role by default...

https://support.pagerduty.com/docs/sso#section-optional-attributes-for-auto-provisioning
>If there is no Role or role attribute in your SAML payload then the user's role will default to the user role. This User role is linked to the **User** basic role and **Manager** advanced permission role.

This certainly looks like the mapping you are referring to.

**Comment by stmcallister** [comment]
Looks like this was resolved by #179. Please repost if that is not the case.
